ang/civicase/ActivitiesCalendar.js

```diff
@@ -433,51 +501,113 @@
+
+      return loadDaysWithActivities(status, date)
+        .then(_.curryRight(updateDaysList)(date)('completed'));
     }
 
     /**
      * Loads the days with at least an incomplete activity
      *
+     * @param {Date} date
      * @return {Promise}
      */
-    function loadDaysWithActivitiesIncomplete () {
-      return loadDaysWithActivities({
-        'IN': CRM.civicase.activityStatusTypes.incomplete
-      })
-        .then(_.curryRight(updateDaysList)('incomplete'));
+    function loadDaysWithActivitiesIncomplete (date) {
+      var status = { 'IN': CRM.civicase.activityStatusTypes.incomplete };
+
+      return loadDaysWithActivities(status, date)
+        .then(_.curryRight(updateDaysList)(date)('incomplete'));
+    }
+
+    /**
+     * Called when the user clicks on a day on the datepicker directive
+     *
+     * If the day has any activities on it, it loads the activities and display
+     * them in a popover
+     */
+    function onDateSelected () {
+      if (!getDayWithActivities($scope.selectedDate)) {
+        return;
+      }
+
+      $scope.loadingActivities = true;
+      $scope.selectedActivites = [];
+
+      $scope.$emit('civicase::ActivitiesCalendar::openActivitiesPopover');
+
+      loadActivitiesOfDate($scope.selectedDate)
+        .then(function (activities) {
+          loadContactsOfActivities(activities)
+            .then(function () {
+              $scope.selectedActivites = activities;
+              $scope.loadingActivities = false;
+            });
+        });
     }
 
     /**
-     * It marks all the days currently in the internal list to be deleted, before
-     * triggering the main load logic
+     * Before calling the main load logic and forcing it to not use the cache, it
+     * performs two type of data reset
      *
-     * This ensures that if some days won't be returned again from any of the
-     * API calls, they will be deleted
+     * hard reset: all months except the one of the currently selected date get
+     * deleted directly from the internal cache
+     *
+     * soft reset: the month of the currently selected date is not deleted, but
+     * its days are marked to be flushed (deleted) later, in case they won't get
+     * returned anymore by the next API requests
+     *
+     * The soft reset avoids removing all the dots at once before even making the
+     * API requests, making for a smoother UI experience
      */
     function reload () {
-      _.each(daysWithActivities, function (day) {
-        day.toFlush = true;
+      var currYearMonth = getYearMonth(selectedDate);
+
+      _.each(daysWithActivities, function (days, yearMonth) {
+        if (yearMonth === currYearMonth) {
+          _.each(daysWithActivities[yearMonth], function (day) {
+            day.toFlush = true;
+          });
+        } else {
+          delete daysWithActivities[yearMonth];
+        }
       });
 
-      load();
+      load(false);
```

Comment:
`load(false)` gives me the impression that the load should not happen. Can we do `load({ useCache: false })` instead?